File: src/main/java/br/com/sistemaos/domain/entity/Endereco.java
package br.com.sistemaos.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Table(name = "endereco")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class Endereco {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "rua", length = 50)
    private String rua;

    @Column(name = "numero", length = 20)
    private String numero;

    @Column(name = "logradouro", length = 80)
    private String logradouro;

    @Column(name = "complemento", length = 200)
    private String complemento;

    //relacionamento
    @OneToOne(mappedBy = "endereco")
    private Cliente cliente;
}

File: src/main/resources/static/app/model/Endereco.js
Ext.define('ProjSistemaOs.model.Endereco', {
    extend: 'Ext.data.Model',
    idProperty: 'id',

    fields: [
        { name: 'id', type: 'int' },
        { name: 'cliente_id', type: 'int' },
        { name: 'rua', type: 'string' },
        { name: 'logradouro', type: 'string' },
        { name: 'numero', type: 'string' },
        { name: 'complemento', type: 'string' }
    ],

    belongsTo: 'Cliente'
});